src/lib/input/input-container.scss

```diff
@@ -211,29 +212,53 @@ $mat-input-underline-disabled-background-image:
   }
 }
 
-// The hint is shown below the underline. There can be more than one; one at the start
-// and one at the end.
-.mat-hint {
-  display: block;
+// Wrapper for the hints and error messages. Provides positioning and text size.
+// Note that we're using `top` in order to allow for stacked children to flow downwards.
+.mat-input-hint-wrapper {
   position: absolute;
   font-size: 75%;
-  bottom: 0;
+  top: 100%;
+  width: 100%;
+  margin-top: -$mat-input-wrapper-spacing;
+}
+
+// The hint is shown below the underline. There can be
+// more than one; one at the start and one at the end.
+.mat-hint {
+  display: block;
+  float: left;
 
   &.mat-right {
-    right: 0;
+    float: right;
   }
 
   [dir='rtl'] & {
-    right: 0;
-    left: auto;
+    float: right;
 
     &.mat-right {
-      right: auto;
-      left: 0;
+      float: left;
     }
   }
 }
 
+// Clears the floats on the hints. Necessary for the `transform` animation to work.
```

Comment:
what exactly is this supposed to do? I don't see any difference if I just remove it.